lib/src/ui/puzzle/puzzle_dashboard_widget.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+
+import 'package:lichess_mobile/src/model/puzzle/puzzle_providers.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+import 'package:lichess_mobile/src/widgets/platform_card.dart';
+import 'package:lichess_mobile/src/model/puzzle/puzzle.dart';
+import 'package:lichess_mobile/src/ui/puzzle/puzzle_dashboard_screen.dart';
+
+class PuzzleDashboardWidget extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final days = ref.watch(daysProvider);
+    final puzzleDashboard =
+        ref.watch(puzzleDashboardProvider(days.toInteger()));
+
+    return puzzleDashboard.when(
+      data: (data) {
+        return ListSection(
```

Comment:
Not sure ListSection will render well on iOS here. But I can deal with that.

Reply:
![simulator_screenshot_A4A8AE36-2F84-45E3-98D2-64DBF041E5DF](https://user-images.githubusercontent.com/54973703/226767447-76f78ebe-57af-442f-aa87-1633fedbd9f9.png)

It looks fine on iPhone 14. Haven't tried other devices